### PR TITLE
Added poetry

### PR DESCRIPTION
**Pull request by jgielstra-cs**
Tweaked tests, linting, pypint, and build scripts for poetry Remove old ModelClient
Added thin Model client for new Model registry

### CODE REVIEW

File: cortex/models.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        Create an MLFlow experiment with default tags
+        :param name: experiment name, must be unique
+        :param tags: optional experiment tags
+        """
+        check_installed()
+        if tags is None:
+            tags = {}
+        # default to client project if project tag isn't specified
+        if tags.get('project') is None:
+            tags['project'] = self._project()
```
Comment (pkandarpa-cs) [inline cortex/models.py:83]: is there a usecase where folks are allowed to override the project to be different from the sensa project?

File: docs/requirements-docs.txt [hunk 2]
```diff
@@ -1,6 +1,6 @@
 Sphinx==5.1.1
 sphinx-multiversion==0.2.4
-GitPython>=3.1.40
+GitPython >=3.1.40
```
Comment (pkandarpa-cs) [inline docs/requirements-docs.txt]: typo?

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+pytest-cov = ">=4.0.0,<5.0.0"
+cuid = "^0.4"
+black = ">=22.6.0,<24"
+sphinx = "^7.2.6"
+gitpython = "^3.1.42"
+sphinx-multiversion = "^0.2.4"
+sphinx-rtd-theme = "^2.0.0"
+sphinxcontrib-restbuilder = "^0.3"
+
+[tool.poetry.extras]
```
Comment (pkandarpa-cs) [inline pyproject.toml:52]: we should version the extras as well. they are versioned in the setup.py below btw

Comment (jgielstra-cs) [reply pyproject.toml:52]: They are versioned above and tagged with optional.  I didn't love this in poetry.  Do you know of a different way ?

File: docs/requirements-docs.txt (hunk 2 above)
Comment (jgielstra-cs) [reply docs/requirements-docs.txt]: Meant to delete this file, yeah typo..

File: cortex/models.py (hunk 1 above)
Comment (jgielstra-cs) [reply cortex/models.py:83]: Project is tied to client, I ?thiink? you can create multiple client object with a project.. but  the lower level methods override it ..  

This is bit gross as the settings are an env variable under covers fro MLFLOW :(

**Pull request merged**